Modules/Sources/BuildSettingsKit/BuildSecrets.swift

```diff
@@ -0,0 +1,100 @@
+public struct BuildSecrets: Sendable {
```

Comment:
Right now, this is no different from `ApiCredentials` (`Secrets.swift`). Eventually, it'll be the interface we use to inject the obfuscated encrypted secrets.

I considered having a dedicated framework for this, but it seemed a waste given BuildSettingsKit already had all the plumbing to access the values.

I also considered putting it in Keystone, but it seems to me having this in BuildSettingsKit will help us extract to smaller more modules in the future.